File: trees/BFS/floodFill.go
func floodFill(image [][]int, sr int, sc int, newColor int) [][]int {
    
    startingColor := image[sr][sc]
    
    queue := [][]int{[]int{sr,sc}}
    
    for (len(queue) > 0) {
        
        // pop from queue
        var popped []int;
        popped, queue = queue[len(queue)-1], queue[:len(queue)-1]
        
        // process the popped
        sr := popped[0]
        sc := popped[1]
        image[sr][sc] = newColor
        
        // for the rest 
        others := [][]int{[]int{1,0}, []int{-1,0}, []int{0,1}, []int{0,-1}}
        for _, set := range others {
            // get item
            sr := set[0] + sr
            sc := set[1] + sc
            
            notOutOfBounds := (sr >= 0 && sr < len(image)) && (sc >= 0 && sc < len(image[0]))
            if notOutOfBounds {
                isStartingColor := (image[sr][sc] == startingColor)
                colorIsNotNewColor := (image[sr][sc] != newColor);
                if isStartingColor && colorIsNotNewColor {
                    queue = append(queue, []int{sr,sc})
                }
            }
        }
    }
    return image
}

File: trees/BFS/floodFill.js
/**
 * @param {number[][]} image
 * @param {number} sr
 * @param {number} sc
 * @param {number} newColor
 * @return {number[][]}
 */
var floodFill = function(image, sr, sc, newColor) {
    var myFloodFill = function(image, sr, sc, newColor, startingColor) {
        
        var queue = [[sr,sc]];
        while (queue.length) {
            var toProcess = queue.pop();
            var sr  = toProcess[0];
            var sc  = toProcess[1];
            
            // do process
            image[sr][sc] = newColor;
            
            // Add the rest to queue
            [[1,0],[-1,0],[0,1],[0,-1]].forEach(set => {
                var sr = toProcess[0] + set[0];
                var sc = toProcess[1] + set[1];
                
                // if it is not out of bounds and is the starting color
                var notOutOfBounds = (sr >= 0 && sr < image.length) && (sc >= 0 && sc < image[0].length);
                if (notOutOfBounds) {
                    var isStartingColor = (image[sr][sc] == startingColor);
                    var colorIsNotNewColor = (image[sr][sc] !== newColor);
                    if (isStartingColor && colorIsNotNewColor) {
                        queue.unshift([sr,sc]);
                    }
                }
            });
        }
        return image;
    }
    
    
    var startingColor = image[sr][sc];
    return myFloodFill(image, sr, sc, newColor, startingColor);
};